### PR TITLE
pd: add missing `dp.eval()` in pd backend

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
Switch to eval mode when evaluating model, otherwise `self.training` will be `True`, backward graph will be created and cause OOM

## Summary by CodeRabbit

- **New Features**
	- Enhanced model evaluation state management to ensure correct behavior during evaluation.

- **Bug Fixes**
	- Improved type consistency in the `normalize_coord` function for better computational accuracy.

### CODE REVIEW

**Review by njzjz — COMMENTED** [review]
Please take a look at @caic99's comment in #4426. Calling `eval()` or `to()` might be expensive and should be considered moving out of the loop.

**Comment by HydrogenSulfate** [comment]
> Please take a look at @caic99's comment in #4426. Calling `eval()` or `to()` might be expensive and should be considered moving out of the loop.

#4426 seems only correct the training state when evaluating during training, but there are still issues in the dp test.

**Comment by caic99** [comment]
> > Please take a look at @caic99's comment in #4426. Calling `eval()` or `to()` might be expensive and should be considered moving out of the loop.
> 
> #4426 seems only correct the training state when evaluating during training, but there are still issues in the dp test.

I did not check if the model has been set to eval state for other entrypoints and backends besides training on PyTorch. Maybe we should fix it for all backends?

**Review by njzjz — APPROVED** [review]
